moveit_ros/planning/kinematics_plugin_loader/src/kinematics_plugin_loader.cpp

```diff
@@ -287,78 +294,71 @@ robot_model::SolverAllocatorFn KinematicsPluginLoader::getLoaderFunction(const s
 
       if (default_solver_plugin_.empty())
       {
-        ROS_DEBUG_NAMED(LOGNAME, "Loading settings for kinematics solvers from the ROS param server ...");
-
-        // read data using ROS params
-        ros::NodeHandle nh("~");
+        RCLCPP_INFO(LOGGER_KINEMATICS_PLUGIN_LOADER,
+                    "Loading settings for kinematics solvers from the ROS param server ...");
 
+        auto ksolver_params = std::make_shared<rclcpp::SyncParametersClient>(node_);
         // read the list of plugin names for possible kinematics solvers
         for (std::size_t i = 0; i < known_groups.size(); ++i)
         {
           std::string base_param_name = known_groups[i].name_;
-          ROS_DEBUG_NAMED("kinematics_plugin_loader", "Looking for param %s ",
-                          (base_param_name + "/kinematics_solver").c_str());
-          std::string ksolver_param_name;
-          bool found = nh.searchParam(base_param_name + "/kinematics_solver", ksolver_param_name);
-          if (!found || !nh.hasParam(ksolver_param_name))
+          std::string ksolver_param_name = base_param_name + "/kinematics_solver";
```

Comment:
This whole part seems overly complicated to me (it already was in the original implementation). But it's probably not the point for premature optimization here. Could you add a `TODO: revise parameter lookup` here so we can track what parameters we need to test/improve in the future?

Reply:
Totally agreed with this point - I added the TODO but I still believe that we should use "." as a delimiter, for example, please take a look at how Navigation2 loads the parameters [yaml](https://github.com/ros-planning/navigation2/blob/6bfa8c48ea0291fff540eb7aead6e5e511130bd7/nav2_bringup/bringup/params/nav2_params.yaml#L113) - [cpp](https://github.com/ros-planning/navigation2/blob/6bfa8c48ea0291fff540eb7aead6e5e511130bd7/nav2_costmap_2d/plugins/obstacle_layer.cpp)

Reply:
I think the yaml you linked doesn't belong to the cpp file. What they did is use [dynamic_reconfigure](https://github.com/ros/dynamic_reconfigure) to generate the parameters: https://github.com/ros-planning/navigation2/blob/6bfa8c48ea0291fff540eb7aead6e5e511130bd7/nav2_costmap_2d/cfg/ObstaclePlugin.cfg. Normal parameters use forward slashes as namespace separators. https://design.ros2.org/articles/ros_parameters.html https://design.ros2.org/articles/topic_and_service_names.html.